vyper/ast_utils.py

```diff
@@ -108,7 +109,11 @@ def parse_to_ast(source_code: str, source_id: int = 0) -> list:
     if '\x00' in source_code:
         raise ParserException('No null bytes (\\x00) allowed in the source code.')
     class_types, reformatted_code = pre_parse(source_code)
-    py_ast = python_ast.parse(reformatted_code)
+    try:
+        py_ast = python_ast.parse(reformatted_code)
+    except SyntaxError as e:
+        # TODO: Ensure 1-to-1 match of source_code:reformatted_code SyntaxErrors
```

Comment:
TODO is here because _theorhetically_ there is a scenario where this code reformatting could affect the location of offsets used to display an error. Would be best if the conversion was accounted for.

Reply:
This is interesting because syntax error reporting actually happens on the reformatted code, but other error reporting can happen on the untransformed code. Look at for instance how the `struct -> class` preparsing rule affects error reporting:
```python
# syntax error
$ vyper -f ir /dev/stdin <<EOF
struct Foo: aoijewofijwef098 \`\`
EOF
  File "<unknown>", line 1
    class Foo: aoijewofijwef098 ``
                                ^
SyntaxError: invalid syntax

During handling of the above exception, another exception occurred:

vyper.exceptions.PythonSyntaxException: line 1:29 SyntaxError: invalid syntax
---> 1 struct Foo: aoijewofijwef098 ``
------------------------------------^

# vs plain structure exception
$ vyper -f ir /dev/stdin <<EOF
struct Foo: aoijewofijwef098   
EOF

Error compiling: /dev/stdin
vyper.exceptions.StructureException: line 1:11 Structs can only contain variables
---> 1 struct Foo: aoijewofijwef098
------------------^
     2
```

Reply:
Does this add buggy behavior now?

Reply:
no, the behavior is as buggy as before

Reply:
... Ship it!